### PR TITLE
Bug 2037075: Add CSI VolumeSource to build types

### PR DESCRIPTION
**Pull request by adambkaplan**
- Augment BuildVolumeSource to inlcude the CSI source type and volume source.
- Re-run `make update` to generate deep copies, conversions, and openapi updates.
- Add validation for build CSI volume sources to ensure that the driver name is set and valid.

### CODE REVIEW

File: pkg/openapi/zz_generated.openapi.go [hunk 1]
```diff
@@ -61597,11 +61602,11 @@ func schema_k8sio_api_core_v1_LimitRangeItem(ref common.ReferenceCallback) commo
...
+							Description: "Type of resource that this limit applies to.\n\nPossible enum values:\n - `\"Container\"` Limit that applies to all containers in a namespace\n - `\"PersistentVolumeClaim\"` Limit that applies to all persistent volume claims in a namespace\n - `\"Pod\"` Limit that applies to all pods in a namespace\n - `\"openshift.io/Image\"` Limit that applies to images. Used with a max[\"storage\"] LimitRangeItem to set the maximum size of an image.\n - `\"openshift.io/ImageStream\"` Limit that applies to image streams. Used with a max[resource] LimitRangeItem to set the maximum number of resource. Where the resource is one of \"openshift.io/images\" and \"openshift.io/image-tags\".",
 							Default:     "",
 							Type:        []string{"string"},
 							Format:      "",
-							Enum:        []interface{}{"Container", "PersistentVolumeClaim", "Pod"}},
+							Enum:        []interface{}{"Container", "PersistentVolumeClaim", "Pod", "openshift.io/Image", "openshift.io/ImageStream"}},
```
Comment (adambkaplan) [inline pkg/openapi/zz_generated.openapi.go:61609]: fyi @dmage @ricardomaraschini

Comment (dmage) [reply pkg/openapi/zz_generated.openapi.go:61609]: LGTM

Comment (dmage) [reply pkg/openapi/zz_generated.openapi.go:61609]: Any ideas why it wasn't there before and what has changed?

Comment (adambkaplan) [reply pkg/openapi/zz_generated.openapi.go:61609]: Odd to see this myself, since openshift/api had been updated recently before this PR merged.

**Review by jkhelil — APPROVED** [review]

**Pull request merged**